packages/@react-stately/color/src/useHexColorFieldState.ts

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {Color} from './Color';
+import {ColorInput, HexColorFieldProps} from '@react-types/color';
+import {NumberFieldState} from '@react-stately/numberfield';
+import {useCallback, useMemo, useState} from 'react';
+import {useControlledState} from '@react-stately/utils';
+
+export interface HexColorFieldState extends NumberFieldState {
+  colorValue: Color,
+  setColorValue: (color: Color) => void,
+  setInputValue: (value: string) => void
```

Comment:
I ended up adding `setInputValue` in the interface even when it's not in the specs #974.

It seems like the colorValue and inputValue works independently.

`colorValue`
* updates in `onChange` as the user types valid colors in hex
* shouldn't capitalize when the user inputs a valid hex color (too distracting?)

`inputValue`
* updates as the user types anything
* becomes capitalized (standard format) on blur

I feel like it violates the independence of the hook though as a react-stately package. It might not make sense why `inputValue` doesn't necessarily change when `colorValue` changes, without context from react-aria.

Reply:
@snowystinger 
just use setColorValue as `string` and do necessary refactor

Reply:
Hi @snowystinger @devongovett 

I ended up keeping `setInputValue` and having `setColorValue` only internal to `useHexColorFieldState`. 
```ts
  colorValue: Color,
  setColorValue: (color: Color) => void,  // internal only
  inputValue: string,  // already provided by NumberFieldState
  setInputValue: (value: string) => void
```

It didn't make sense to use `colorValue` and `setColorValue` for `string` types so I used `inputValue` and `setInputValue` instead. I still needed a pair like `colorValue` and `setColorValue` to handle the actual `Color` data we have.